CapstoneProject/Assets/Scripts/Core/Controller/Dialogue/DialogueManager.cs

```diff
@@ -197,33 +213,177 @@ private void BeginText()
+                }
+                UnityEngine.Debug.Log("Finished drawing text");
+                StopCoroutine(textCoroutine);
+            }
+
+            if (dialogue == "maindialogue1" && dialogueState != 1)
+            {
+
+            }
+
+            else if (dialogue == "maindialogue1" || dialogue == "maindialogue2" || dialogue == "maindialogue3" || dialogue == "maindialogue4" || dialogue == "playerdeath")
+            {
+                dialogueState++;
+                UnityEngine.Debug.Log("Dialogue state increased");
+            }
+
+            
+
+            else if (!dialogueSkipRepeat)
+            {
+                UnityEngine.Debug.Log("Drawing text");
+                textCoroutine = TextScrollInput(dialogue);
+                dialoguePriority = true;
+                StartCoroutine(textCoroutine);
+                while (dialoguePriority)
+                {
+                    yield return new WaitForSeconds(0.25F);
+                }
+                UnityEngine.Debug.Log("Finished drawing text");
+                StopCoroutine(textCoroutine);
+            }
+        }
+
+        yield return null;
+    }
+
+    private IEnumerator TextScrollInput(string displayText)
+    {
+        IEnumerator textCoroutine = TextScroll(displayText);
+        StartCoroutine(textCoroutine);
+        UnityEngine.Debug.Log("In the input enumerator");
+
+        bool input = false;
+        while (!input)
+        {
+            if (Input.GetKeyDown(KeyCode.E))
+            {
+                input = true;
+                dialoguePriority = false;
+            }
+            yield return null;
+        }
+        StopCoroutine(textCoroutine);
     }
 
-    IEnumerator TextScroll(string displayText)
+    private IEnumerator SearchForState(IEnumerator textCoroutine, string searchText, string checkText) 
     {
+        foreach (string dialogueStatus in loadedDialogue)
+        {
+            if (dialogueStatus == searchText)
+            {
+                int position = loadedDialogue.IndexOf(dialogueStatus);
+                foreach (string dialogueStatusConfirmed in loadedDialogue)
+                {
+                    if (dialogueStatusConfirmed == checkText)
+                    {
+                        break;
+                    }
+                    if (loadedDialogue.IndexOf(dialogueStatusConfirmed)! < position)
+                    {
+                        UnityEngine.Debug.Log("Drawing text");
+                        textCoroutine = TextScrollInput(dialogueStatusConfirmed);
+                        dialoguePriority = true;
+                        StartCoroutine(textCoroutine);
+                        while (dialoguePriority)
+                        {
+                            yield return new WaitForSeconds(0.25F);
+                        }
+                        UnityEngine.Debug.Log("Finished drawing text");
+                        StopCoroutine(textCoroutine);
```

Comment:
I don't think this stop is required and could maybe lead to a race condition bug.